### PR TITLE
ci: try fix format in CI

### LINKED ISSUE #8636: Run fix_format in ci and produce diff
We're adding more stuff in the format CI and it become heavier and with more dependencies to run. We can produce a diff in azp in case of check_format fails, so the diff gives what CI expect for the fixable format without environment dependent bit.

@htuch @jmarantz

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
Risk Level:
Testing: testing CI
Docs Changes:
Release Notes:
Fixes #8636

### CODE REVIEW

File: api/envoy/api/v3alpha/core/base.proto [hunk 1]
```diff
@@ -244,9 +244,7 @@ message TransportSocket {
 
   // Implementation specific configuration which depends on the implementation being instantiated.
   // See the supported transport socket implementations for further documentation.
-  oneof config_type {
-    google.protobuf.Any typed_config = 3;
-  }
+  google.protobuf.Any typed_config = 3;
```
Comment (lizan) [inline api/envoy/api/v3alpha/core/base.proto]: To test CI actually produces diff, will revert once it is tested.

File: ci/check_and_fix_format.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+set -e
+
+function fix {
+  set +e
+  ci/do_ci.sh fix_format
+  ci/do_ci.sh fix_spelling
+  ci/do_ci.sh fix_spelling_pedantic
+  echo "Format check faild, try apply following patch to fix:"
+  git diff HEAD | tee /build/fix_format.diff
```
Comment (jmarantz) [inline ci/check_and_fix_format.sh]: WDYT of allowing a command-line override of the location of the generated diff file so we can run this outside of docker?

File: ci/check_and_fix_format.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  exit 1
+}
+
+trap fix ERR
```
Comment (jmarantz) [inline ci/check_and_fix_format.sh]: i'm unclear on what this is doing (and I thought i knew bash a little :) can you just add some comments?

File: ci/check_and_fix_format.sh (hunk 2 above)
Comment (lizan) [reply ci/check_and_fix_format.sh]: Added env var override, though `ci/do_ci.sh` is hard to run outside of docker anyway.

File: ci/check_and_fix_format.sh (hunk 3 above)
Comment (lizan) [reply ci/check_and_fix_format.sh]: added.

File: ci/check_and_fix_format.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  ci/do_ci.sh fix_format
+  ci/do_ci.sh fix_spelling
+  ci/do_ci.sh fix_spelling_pedantic
+  echo "Format check faild, try apply following patch to fix:"
+  git diff HEAD | tee "${DIFF_OUTPUT}"
+
+  exit 1
+}
+
+# If any of check fails, run fix function above.
```
Comment (jmarantz) [inline ci/check_and_fix_format.sh]: nit: "if any of the checks fail, run the fix function above."

**Pull request merged**